packages/accounts/src/light-account/erc1271/erc1271.ts

```diff
@@ -0,0 +1,97 @@
+import type {
+  Address,
+  SmartAccountSigner,
+  ToSmartContractAccountParams,
+} from "@alchemy/aa-core";
+import { hashDomain } from "./utils/hashDomain.js";
+import { concat, hashMessage, hashTypedData } from "viem";
+import { hashType, hashStruct } from "./utils/hashTypedData.js";
+
+export interface GetErc1271SigningFunctionsParams {
+  accountAddress: Address;
+  accountName: string;
+  accountVersion: string;
+  chainId: number;
+  signer: SmartAccountSigner;
+  wrapperTypeName: string;
+}
+
+export const getErc1271SigningFunctions = ({
+  accountAddress,
+  accountName,
+  accountVersion,
```

Comment:
Version is always 2 here so no need for the version input. Might want to guard only for version 2 here

Reply:
I'm trying to decouple these functions from light account, since this signing mechanism is a general proposal that other accounts can use. That's also the reason why I don't prepend the `SignatureType` enum in these functions but instead have the light account impl call these and prepend afterwards.

Reply:
Wondering if it's worth fetching this and other domain separator values from `account.eip712Domain()` (if that context is readily available from somewhere). From 2.0.0, we support https://eips.ethereum.org/EIPS/eip-5267.

Reply:
I could be wrong, but my feeling is doing that would be less secure. The case I'm worried about is where you call the SDK's `createLightAccount` function but you pass it an account address which is not a `LightAccount`, either by mistake or because you were tricked by a bad actor. Then you would end up signing messages with the domain separator reported by that account, which could run into the replay attack we were trying to avoid in the first place if that domain separator is for a different SCA that shares an owner with the light account.

Reply:
> The case I'm worried about is where you call the SDK's createLightAccount function but you pass it an account address which is not a LightAccount

I'm still getting caught up on the changes, but in `main` the way LightAccount handles signing is by fetching the LightAccount version using the on-chain implAddress. This way, if you pass in an address that isn't LA, you would end up getting an error

Reply:
> I'm trying to decouple these functions from light account

nit: should we then move this into a `utils/signing` folder either in `core` or here in `accounts`?

Reply:
> fetching the LightAccount version using the on-chain implAddress
That's a good idea, but I just looked up how it's implemented and the way it's written now it's good as a check against mistakes but doesn't provide security against malicious contracts, since it's looking up the impl storage slot and a malicious contract can put whatever it wants there.

If you wanted security you'd need to check the proxy codehash too, which I maybe wouldn't recommend since it could reject other proxy impls that have upgraded to light account.